tests/FakeItEasy.Specs/OrderedCallMatchingSpecs.cs

```diff
@@ -1,169 +1,210 @@
 namespace FakeItEasy.Specs
 {
     using System;
+    using FakeItEasy.Configuration;
     using FakeItEasy.Tests.TestHelpers;
     using FluentAssertions;
     using Xbehave;
 
     public static class OrderedCallMatchingSpecs
     {
-        public interface IHaveNoGenericParameters
+        public interface IFoo
         {
             void Bar(int baz);
         }
 
-        public interface IHaveOneGenericParameter
+        public interface ISomething
         {
-            void Bar<T>(T baz);
+            void SomethingMethod();
+        }
+
+        public interface ISomethingBaz : ISomething
+        {
+            void BazMethod();
+        }
+
+        public interface ISomethingQux : ISomething
+        {
+            void QuxMethod();
         }
 
         [Scenario]
-        public static void NonGenericCallsSuccess(
-            IHaveNoGenericParameters fake,
-            ISequentialCallContext sequentialCallContext,
-            Exception exception)
+        public static void OrderedAssertionsInOrder(IFoo fake, Exception exception)
         {
             "Given a Fake"
-                .x(() => fake = A.Fake<IHaveNoGenericParameters>());
+                .x(() => fake = A.Fake<IFoo>());
+
+            "And a call on the Fake, passing argument 1"
+                .x(() => fake.Bar(1));
 
             "And a call on the Fake, passing argument 1"
                 .x(() => fake.Bar(1));
 
             "And a call on the Fake, passing argument 2"
                 .x(() => fake.Bar(2));
 
-            "And a call-ordering context"
-                .x(c => sequentialCallContext = A.SequentialCallContext());
+            "And a call on the Fake, passing argument 3"
+                .x(() => fake.Bar(3));
 
-            "When I expect the call with argument 1 to have been made"
-                .x(() => A.CallTo(() => fake.Bar(1)).MustHaveHappened().InOrder(sequentialCallContext));
+            "When I expect a call with argument 1 to have been made twice and then a call with argument 2 and then a call with argument 3"
+                .x(() => exception = Record.Exception(() =>
+                    A.CallTo(() => fake.Bar(1)).MustHaveHappened(Repeated.Exactly.Twice)
+                        .Then(A.CallTo(() => fake.Bar(2)).MustHaveHappened())
+                        .Then(A.CallTo(() => fake.Bar(3)).MustHaveHappened())));
 
-            "And I expect the call with argument 2 to have been made next"
-                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(2)).MustHaveHappened().InOrder(sequentialCallContext)));
-
-            "Then the assertion should succeed"
-                .x(() => exception.Should().BeNull("because the assertion should have succeeded"));
+            "Then the assertion should pass"

```

Comment:
Yes. I did not do a good job with this. The documentation calls things "assertions". I will make everything like this.